### ISSUE TITLE
Allow modification of source attributes while using sourcekey

### ISSUE THREAD

**Issue by RFSH** [informatics-isi-edu/ermrestjs#909]
Currently if `sourcekey` is used to refer to any of the defined `source-definitions`, the rest of "source attributes" will be ignored. By "source attributes" I'm referring to the attributes that change the definition of a pseudo-column (basically change the generated hash). These attributes are: `source`, `entity`, `aggregate`, and `self_link`. 

- [x] `source` should still be ignored if defined alongside `sourcekey`, but `entity`, `aggregate`, and `self_link` should be allowed. This would allow data-modelers to define a path once and then use `aggregate` in compact while showing the inline table in detailed (or showing it in facet without `aggregate`):

```javascript
"tag:isrd.isi.edu,2019:source-definitions": {
  "sources": {
    "S_taxonomy": {
      "markdown_name": "Taxonomy",
      "source": [
        {"sourcekey": "S_core_fact"},
        {"inbound": ["CFDE", "core_fact_ncbi_taxonomy_core_fact_fkey"]},
        {"outbound": ["CFDE", "core_fact_ncbi_taxonomy_ncbi_taxon_fkey"]},
        "nid"
      ]
    }
  }
},
"tag:isrd.isi.edu,2016:visible-columns": {
  "compact": [
       {"sourcekey": "S_taxonomy", "aggregate": "array_d"}
  ],
  "detailed": [
    {"sourcekey": "S_taxonomy"}
  ],
 "filter": {
   "and": [
     {"sourcekey": "S_taxonomy"}
   ]
 }
}
```

- [ ] If we allow this, to make the definition simpler, we might also want to allow sources to refer to each other using `sourcekey`:

```javascript
"tag:isrd.isi.edu,2019:source-definitions": {
  "sources": {
    "S_taxonomy": {
      "markdown_name": "Taxonomy",
      "source": [
        {"sourcekey": "S_core_fact"},
        {"inbound": ["CFDE", "core_fact_ncbi_taxonomy_core_fact_fkey"]},
        {"outbound": ["CFDE", "core_fact_ncbi_taxonomy_ncbi_taxon_fkey"]},
        "nid"
      ]
    },
    "S_taxonomy_array": {
      "sourcekey": "S_taxonomy",
      "aggregate": "array_d"
    }
  }
}
```

**Comment by RFSH** [comment]
As part of the changes in [this PR](https://github.com/informatics-isi-edu/ermrestjs/pull/992) I initially pushed a change that would allow the usage of `sourcekey` inside the `source-definitions` annotation. I pushed a simple syntactic sugar to avoid copy/paste. As a result, when this source was used in a `visible-column`, there was no signal that it's using a different sourcekey. Therefore, it couldn't do the path-sharing logic correctly. 

That's why we decided to revert that change and discuss this later when we needed this functionality.
